src/common/exporter/word/exporters/outline.js

```diff
@@ -44,15 +45,24 @@ export default function exportOutline(state, namesMapping, doc, options) {
     const uniqueBeatTitleSelector = makeBeatTitleSelector(state)
     const title = uniqueBeatTitleSelector(state, beat.id)
     let paragraphs = [new Paragraph('')]
-    paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
+    if (!outlineFilter) {
+      paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
+    }
 
     if (options.outline.sceneCards) {
       const cards = beatCardMapping[beat.id]
       const customAttrs = cardsCustomAttributesSelector(state)
       const sortedCards = sortCardsInBeat(beat.autoOutlineSort, cards, lines)
-      let cardParagraphs = sortedCards.flatMap((c) =>
-        card(c, linesById, namesMapping, customAttrs, doc, options)
-      )
+      let flag = 0
+      const cardParagraphs = sortedCards.flatMap((c) => {
+        if (outlineFilter) {
+          if (!outlineFilter.includes(c.lineId)) return
+          flag++
+          if (flag === 1)
+            paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
+        }
+        return card(c, linesById, namesMapping, customAttrs, doc, options)
+      })
```

Comment:
It seems to me that we want to add a level two-paragraph when we have at least one card that wasn't filtered out(?)
This code as it stands is trying to do too many things at the same time.

This has resulted in a bug: if you return from the body of a call to `flatMap` with undefined (line 59) then you'll get an undefined in an array (e.g. `[x, y, ...undefined, ..., z]`)

If we instead separate things then the problem becomes simpler.
Let's start by getting a collection of cards that aren't filtered,
```javascript
  const filteredCards = !outlineFilter ? sortedCards : sortedCards.filter((card) => {
    outlineFilter.includes(card.lineId)
  }
```
then let's add the level 2 paragraph if there are cards,

```javascript
  if (filteredCards.length) {
    paragraphs.push(new Paragraph({ text: title, heading: HeadingLevel.HEADING_2 }))
  }
```

finally, let's add the paragraphs for each card.
```javascript
  const cardParagraphs = filteredCards.flatMap((c) =>
    card(c, linesById, namesMapping, customAttrs, doc, options)
  )
```